src/Gui/ViewProviderPyImp.cpp

```diff
@@ -60,33 +60,56 @@ std::string ViewProviderPy::representation() const
     return "<View provider object>";
 }
 
-PyObject*  ViewProviderPy::addProperty(PyObject *args)
-{
-    char *sType,*sName=nullptr,*sGroup=nullptr,*sDoc=nullptr;
-    short attr=0;
-    std::string sDocStr;
-    PyObject *ro = Py_False, *hd = Py_False;
-    if (!PyArg_ParseTuple(args, "s|ssethO!O!", &sType,&sName,&sGroup,"utf-8",&sDoc,&attr,
-        &PyBool_Type, &ro, &PyBool_Type, &hd))
+PyObject* ViewProviderPy::addProperty(PyObject* args, PyObject* kwd)
+{
+    char *sType, *sName = nullptr, *sGroup = nullptr, *sDoc = nullptr;
+    short attr = 0;
+    PyObject *ro = Py_False, *hd = Py_False, *lk = Py_False;
+    PyObject* enumVals = nullptr;
+    const std::array<const char*, 10> kwlist {"type",
+                                              "name",
+                                              "group",
+                                              "doc",
+                                              "attr",
+                                              "read_only",

```

Comment:
"read_only" in .pyi file is written as "readOnly" - they must be in sync, I think it would be best to update .pyi file.